projects/packages/forms/src/blocks/field-select/edit.js

```diff
@@ -63,22 +62,12 @@ export default function DropdownFieldEdit( props ) {
 			templateLock: 'all',
 		}
 	);
-
-	// Note: Border styles aren't rendered on the frontend select dropdown so are
-	// omitted from the styles applied to the dropdown options wrapper here.
-	const inputColorStyles = getColorClassesAndStyles( inputBlockAttributes ?? {} );
-	const inputTypographyStyles = getTypographyClassesAndStyles( inputBlockAttributes ?? {} );
+	const inputStyles = useFormStyleOutlineClassesAndStyles( {
```

Comment:
The way we're using both inline styles and CSS vars it feels like there's a lot of duplication.

Random idea - but I wonder if we can move away from needing to call this hook on the field wrappers at all.

It looks like we're only using this to set two css vars (and then also `--notch-width` which is derived from those), is that right?

These css vars aren't used by the field wrapper at all from what I can tell, they're only used by the label and input.

The label already calls this hook, so it knows the border values. The input also knows the border values. We could switch to setting any vars on the label and input instead.

Maybe that provides an opportunity to simplify things.

Reply:
Thanks for looking at this.

There's a lot that can be refactoring but I'm still struggling with getting 1:1 basic functionality.

From memory, we have to set the CSS vars at a higher level so child elements can use their default values, e.g., notched elements needs the border widths as well to calculate padding.

Please feel free to refactor this how you think is best - I'd appreciate the help on this one as I'm slowly losing the overview given how many rabbit holes there are.